### DISCUSSION TITLE
How to use data augmentation to increase number of training samples

### DISCUSSION THREAD

**Discussion by Jazzzzie** [facebookresearch/detectron2#2813] [Q&A]
I understand how to write a custum_mapper funciton to perform data augmentation on the registered training dataset, but I have a few questions:

- How can I visually inspect the results of the transformation steps?

Currently, I create a dataset_dict and metadataset and use those for showing the training images. 

The custom_mapper function takes a dataset-dict as input, and returns one as well, but the exact type and structure are different. (in the input, there is "annotation", in the output "instances", and in different format, which doesnt allow plotting the image and annotations as the input dataset_dict can.

- How can I use data augmentation to also increase the number of training images?

Performing the transformations is nice, but I want to use both the raw input, as the rotated/flipped etc. images.

I use instance segmentation with polygon annotated instances.

**Comment by Jazzzzie** [comment]
ended up using [https://github.com/joheras/CLoDSA](https://github.com/joheras/CLoDSA)

**Reply by slgxmh** [reply]
Thanks for you sharing, I have the same problem. Are you use online augmentation online in the mapper function, or use CLoDSA to augment offline with saved in files.
And if you use online augmentation, may you share codes of the mapper function?

**Reply by Jazzzzie** [reply]
I used it offline with local files
